### PR TITLE
TwacodeRender mention, mail, link color change 

### PR DESCRIPTION
**Pull request by EvGen94**
#753
Added unique colors for mentions equal to the color of the user's name in the chat, changing colors for links and mails depending on the message sender (own or not)

### CODE REVIEW

File: lib/utils/twacode.dart [hunk 1]
```diff
@@ -650,7 +654,7 @@ class TwacodeRenderer {
 
       case TType.User:
         style = TextStyle(
-          color: Colors.lightBlue,
+          color: HSLColor.fromAHSL(1, userUniqueColor!, 0.9, 0.3).toColor(),
```
Comment (bobs4462) [inline lib/utils/twacode.dart:657]: Why not to make the field required?

File: lib/utils/twacode.dart [hunk 2]
```diff
@@ -700,7 +720,9 @@ class TwacodeRenderer {
   }
 
   List<InlineSpan> render(
-      {required List<dynamic> twacode, TextStyle? parentStyle}) {
+      {required List<dynamic> twacode,
+      TextStyle? parentStyle,
+      double? userUniqueColor}) {
```
Comment (bobs4462) [inline lib/utils/twacode.dart:725]: Same here, either make it required, or provide a default

**Review by bobs4462 — CHANGES_REQUESTED** [review]
I think if we are sure that the uniqueColor field is always present, let's just make it required

File: lib/utils/twacode.dart (hunk 2 above)
Comment (EvGen94) [reply lib/utils/twacode.dart:725]: You're right, I will go through the entire file to check all the fields then for consistency

File: lib/utils/twacode.dart [hunk 3]
```diff
@@ -8,86 +8,86 @@ import 'package:url_launcher/url_launcher.dart';
...
 
   TwacodeParser(this.original) {
-    parse();
+    parse(original);
   }
 
   List<dynamic> get message => nodes.map((n) => n.transform()).toList();
 
-  void parse() {
+  void parse(String? original) {
```
Comment (bobs4462) [inline lib/utils/twacode.dart:21]: what if it's null?

**Review by bobs4462 — CHANGES_REQUESTED** [review]

File: lib/utils/twacode.dart (hunk 3 above)
Comment (bobs4462) [reply lib/utils/twacode.dart:21]:
```dart
void parse([String original = ''])
```
Just make it like that

**Review by bobs4462 — CHANGES_REQUESTED** [review]

**Review by bobs4462 — APPROVED** [review]

**Pull request merged**